### PR TITLE
Return the port the WebUI is bound to (useful if port 0 was used)

### PR DESCRIPTION
**Pull request by alig**

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/deploy/DeployMessage.scala [hunk 1]
```diff
@@ -127,4 +127,8 @@ private[deploy] object DeployMessages {
 
   case object CheckForWorkerTimeOut
 
+  case object RequestWebUIPort
+
+  case class WebUIPortResponse(webUIBoundPort: Int) {}
```
Comment (rxin) [inline core/src/main/scala/org/apache/spark/deploy/DeployMessage.scala]: You can remove the {}

File: core/src/main/scala/org/apache/spark/deploy/master/Master.scala [hunk 2]
```diff
@@ -26,6 +26,8 @@ import akka.actor._
 import akka.actor.Terminated
 import akka.remote.{RemoteClientLifeCycleEvent, RemoteClientDisconnected, RemoteClientShutdown}
 import akka.util.duration._
+import akka.dispatch.Await
```
Comment (rxin) [inline core/src/main/scala/org/apache/spark/deploy/master/Master.scala:27]: sort the imports in the following order:
1. java
2. scala
3. all other packages alphabetic order.
4. org.apache.spark

**Comment by alig** [comment]
Jenkins, please test this

File: core/src/main/scala/org/apache/spark/deploy/master/Master.scala [hunk 3]
```diff
@@ -378,9 +386,12 @@ private[spark] object Master {
     }
   }
 
-  def startSystemAndActor(host: String, port: Int, webUiPort: Int): (ActorSystem, Int) = {
+  def startSystemAndActor(host: String, port: Int, webUiPort: Int): (ActorSystem, Int, Int) = {
     val (actorSystem, boundPort) = AkkaUtils.createActorSystem(systemName, host, port)
     val actor = actorSystem.actorOf(Props(new Master(host, boundPort, webUiPort)), name = actorName)
-    (actorSystem, boundPort)
+    implicit val timeout = Timeout(5 seconds)
```
Comment (rxin) [inline core/src/main/scala/org/apache/spark/deploy/master/Master.scala]: Actually let's reuse an existing akka timeout setting. See 

https://github.com/mesos/spark/blob/master/core/src/main/scala/org/apache/spark/storage/BlockManagerMasterActor.scala#L53

**Comment by mateiz** [comment]
Looks good, thanks.

**Pull request merged**